core/src/main/scala/cats/ApplicativeError.scala

```diff
@@ -74,6 +76,34 @@ trait ApplicativeError[F[_], E] extends Applicative[F] {
   def recoverWith[A](fa: F[A])(pf: PartialFunction[E, F[A]]): F[A] =
     handleErrorWith(fa)(e =>
       pf applyOrElse(e, raiseError))
+  /**
+   * Often E is Throwable. Here we try to call pure or catch
+   * and raise.
+   */
+  def catchNonFatal[A](a: => A)(implicit ev: Throwable <:< E): F[A] =
+    try pure(a)
+    catch {
+      case NonFatal(e) => raiseError(e)
+    }
+
+  /**
+   * Often E is Throwable. Here we try to call pure or catch
+   * and raise
+   */
+  def catchNonFatalEval[A](a: Eval[A])(implicit ev: Throwable <:< E): F[A] =
```

Comment:
I think that would be slightly slower since you would allocate another closure to do the call by name.